### PR TITLE
Ability to add icon

### PR DESCRIPTION
**Pull request by fernandodebrando**
I made an improvement that enables add icon to breadcrumbs.

### CODE REVIEW

File: src/Creitive/Breadcrumbs/Breadcrumbs.php [hunk 1]
```diff
@@ -350,6 +356,35 @@ public function setListElement($element)
...
+        if (!is_string($classIcon) && !is_null($classIcon))
+        {
+            throw new \InvalidArgumentException(
+                'Breadcrumbs::setIcon() only accepts strings or NULL, but '
+                . (is_object($classIcon) ? get_class($classIcon) : gettype($classIcon))
+                . ' given: ' . print_r($classIcon, true)
+            );
+        }
+
+        $this->icon = " <i class='{$classIcon}'></i> ";
```
Comment (levacic) [inline src/Creitive/Breadcrumbs/Breadcrumbs.php]: I know it's not the most readable, but the rest of the library uses string concatenation and double quoted attributes within the rendered output, so could you change this to use the same convention?

Also, I don't know about the space before the icon, maybe that kind of spacing should be handled by the end user via CSS? I don't have strong feelings about it either way.

Comment (levacic) [inline src/Creitive/Breadcrumbs/Breadcrumbs.php]: Also, what do you think about simply storing the actual passed icon in `$this->icon`? Then `setIcon()` and `getIcon()` would be symmetrical, whereas right now, `getIcon()` returns something notably different than what was passed into `setIcon()`.

Also, right now, if `null` is passed, `$this->icon` will be a string containing an empty `<i>` element with no class set.

File: src/Creitive/Breadcrumbs/Breadcrumbs.php [hunk 2]
```diff
@@ -473,6 +508,7 @@ public function render()
 
 		$cssClasses = implode(' ', $this->breadcrumbsCssClasses);
 		return '<'. $this->listElement .' class="' . $cssClasses .'">' 
+		       . $this->icon
```
Comment (levacic) [inline src/Creitive/Breadcrumbs/Breadcrumbs.php]: I think this is where we should do the actual "rendering", depending on whether or not `$this->icon` is `null`.

**Comment by levacic** [comment]
@fernandodebrando Thank you for the PR, and I'm terribly sorry for the unresponsiveness. I've commented some lines I feel should be updated.

**Comment by fernandodebrando** [comment]
@levacic great comments, i am submitting a new pull request with the changes.

**Comment by levacic** [comment]
Hi @fernandodebrando, thanks for the update, this looks better!

Can you just fix the indentation so tabs are used everywhere?

Also, about your icon styling suggestion, mind adding it to the documentation?

I'll migrate the project to PSR-2 once this PR and #15 are both merged, but for now, let's stay consistent.

**Comment by levacic** [comment]
Also, could you add some unit tests? I think it would be useful to test that there is no icon in the rendered output when `setIcon(null)` is called, and that the expected icon is there after calling `setIcon('foo')`.

File: src/Creitive/Breadcrumbs/Breadcrumbs.php [hunk 3]
```diff
@@ -350,6 +356,35 @@ public function setListElement($element)
 		$this->listElement = $element;
 	}
 
+/**
```
Comment (levacic) [inline src/Creitive/Breadcrumbs/Breadcrumbs.php]: Missing a tab here.

File: src/Creitive/Breadcrumbs/Breadcrumbs.php [hunk 4]
```diff
@@ -473,6 +508,7 @@ public function render()
 
 		$cssClasses = implode(' ', $this->breadcrumbsCssClasses);
 		return '<'. $this->listElement .' class="' . $cssClasses .'">' 
+		       . (!is_null($this->icon) ? '<i class="'. $this->icon .'"></i>' : '')
```
Comment (levacic) [inline src/Creitive/Breadcrumbs/Breadcrumbs.php]: Hey, I've just noticed - this actually adds an `<i>` element as a direct child of an `<ul>` (or `<ol>` if the users sets that) - which isn't valid. So we have to figure out a different strategy to accomplish this - do you have any ideas what the best way to do that would be?

**Comment by levacic** [comment]
We haven't had any activity here in a while, and upon further examination, it's probably just cleaner to add this via CSS. I'll close the PR, sorry for the trouble!
